### ISSUE TITLE
Choose multiple servers and render graphs for each

### ISSUE THREAD

**Issue by dc7707** [alexjustesen/speedtest-tracker#1452]
**Is your feature request related to a problem? Please describe.**
It's not so much a problem, but it could be useful to see results from multiple servers to rule out one server being slow at the moment of testing.

**Describe the solution you'd like**
An option to test multiple servers in the settings.
A box to select the servers like the existing one for using one at a time.
When enabled and selected a speedtest is ran against every selected server (with an apropriate delay so they don't get in each other's way), and graphs are rendered for each.
When looking at the graphs one could get a good idea of their performance by averaging them. (This could also be calculated and displayed in some way.)

**Additional context**
I know this isn't a big deal for most people, but I think it would be cool. Thank you for taking the time to read this request, and thank you for this wonderful tool!

**Comment by bogdanr** [comment]
Agree. This would be very useful. By default speedtest uses a server in the provider's network which is not a good measure of Internet speed but actually how congested is the provider's network. If I chose a server from another country I could see Internet speed, but I would like to have both.

**Comment by fvdm** [comment]
Yeah this is much needed. As @bogdanr points out the major downside of pretty much all WAN speedtests is that you are actually testing the peering instead of your own connection. When they have issues your speedtest will come out bad.

I often get notifications that the speed it under the threshold. But it nearly always is that the selected server is having network problems. By using multiple servers Speedtest Tracker could use the highest bandwidth and lowest ping for the end result.

Having stats per server can be useful for cleaning up the custom servers list. Maybe automatic, exclude the bad ones for a day?

**Comment by ComputerGuy99** [comment]
Hello everyone,
I starred this issue a while back because I wanted to see how good my ISPs international peering is. After looking around the internet for a while I did not find any existing solution for running speed tests against multiple servers and graphing them so I ended up building my own. You can check out my code as well as a sample map I created here: https://github.com/ComputerGuy99/global-internet-speed-test  I would love to hear your feedback on my solution.

**Comment by Nicba1010** [comment]
I could really use this, as I have problems with my ISP speeds and trying to collect proof to show them. To the closest speedtest servers it works fine (as they're hosted by the ISP) but as soon as it leaves their network it drops like a rock, from 900/450 to 5/150...
